source/building-your-own-javascript-components/index.html.md.erb

```diff
@@ -0,0 +1,283 @@
+    Component.checkSupport()
+
+    // Check for support of the native clipboard API
+    if (!navigator.clipboard) {
+      throw new Error(‘Clipboard API not supported’)	
+    }
+  }
+}
+```
+## Using `createAll` with your components
+
+Use the `createAll` function to initialise your components [the same way GOV.UK Frontend components are initialised](../import-javascript/#import-individual-components). The function will look up all the HTML elements on which a given component should be initialised and initialise a component for each of them.
+
+```js
+import {createAll} from ‘govuk-frontend’
+import {ProjectComponent, OtherProjectComponent} from ‘./project-components.js’
+
+createAll(ProjectComponent)
+// You can provide a config for components that use them
+createAll(OtherProjectComponent, {
+  anOption: ‘itsValue’
+})
+```
+
+The `createAll` function also lets you:
+
+- [only search a specific part of the page for elements](#initialise-only-on-part-of-a-page) (for example, after adding new content with JavaScript)
+- [respond to errors during components initialisation](#handling-initialisation-errors) (for example, sending them to an error reporting service)
+
+To avoid reimplementing features shared across components, we recommend that your components extend [our `Component` class](#using-the-component-class). However, this is not a requirement, and all a component needs to be used with `createAll` is:
+
+- a [`moduleName` static property](#defining-the-modulename-property)
+- a constructor expecting an HTML element as a first parameter and any necessary configuration options as a second parameter
+
+### Initialise only on part of a page
+
+By default, `createAll` looks through the whole page for elements to initialise the components on. If you update a page with new markup, (for example, a modal dialog box), you can initialise components on that part of the page only.
+
+```js
+import {createAll} from ‘govuk-frontend’
+import {ProjectComponent} from ‘./project-component.js’
+
+const $element = document.querySelector(‘.app-dialog’)
+
+createAll(ProjectComponent, {SOME_CONFIGURATION}, $element)
+// If the component does not need any configuration, pass `null` for the configuration
+createAll(ProjectComponent, null, $element)
+```
+
+### Handling initialisation errors
+
+By default, `createAll` catches errors thrown by components during their initialisation and logs them in the console. This makes sure components further down the page still get initialised after an error. You may still want to respond to errors happening during component initialisation (for example, notifying an error monitoring service) without [manually initialising each component](../import-javascript/#import-individual-components). 
+
+You can use a function as a third argument to respond to errors being thrown while components are being initialised. If a component throws an error, the function will be called and will receive:
```

Comment:
Disregard this one too 'throw' and 'catch' are indeed commonly understood coding terms and describe specific parts of the error handing process that these suggested terms don't capture.

```suggestion
You can use a function as a third argument to respond to errors caused while components are being initialised. If a component throws an error, the function will be called and will receive:
```